backend/functions/src/triggers/on-update-contract.ts

```diff
@@ -113,6 +119,46 @@ const getPropsThatTriggerRevalidation = (contract: Contract) => {
   }
 }
 
+async function updateContractSubcollectionsVisibility(
+  contractId: string,
+  newVisibility: 'public' | 'unlisted'
+) {
+  const contractRef = firestore.collection('contracts').doc(contractId)
+
+  // Update comments' visibility
+  const commentsRef = contractRef.collection('comments')
+  await processSubcollection(commentsRef, newVisibility)
+
+  // Update bets' visibility
+  const betsRef = contractRef.collection('bets')
+  await processSubcollection(betsRef, newVisibility)
+}
+
+async function processSubcollection(
+  subcollectionRef: FirebaseFirestore.CollectionReference,
+  newVisibility: 'public' | 'unlisted'
+) {
+  const batchSize = 500
+
+  let moreDocuments = true
+
+  while (moreDocuments) {
+    const snapshot = await subcollectionRef.limit(batchSize).get()
```

Comment:
Needs an offset or it would always keep fetching the same docs. See `processPaginated` in backend/shared/utils.ts